### PR TITLE
Split the cwfs module into more descriptive modules

### PR DESCRIPTION
**Pull request by jfcrenshaw**
This is the second PR towards merging the new implementation of the TIE into ts_wep. This PR just splits the CWFS module (which was originally a port of Bo's CWFS code) into smaller, more descriptively named modules.

The PR touches a lot of files, because it
(1) Splits `lsst.ts.wep.cwfs` into `lsst.ts.wep.centroid`, `lsst.ts.wep.templates`, and `lsst.ts.wep.estimation`
(2) Changes the corresponding imports in the other modules
(3) Moves the corresponding unit tests to reflect this restructuring
(4) Changes the documentation to reflect the new modules

However, this PR does not make any changes to the underlying code or unit tests -- it merely moves the existing code and tests around, so that the changes made by the big TIE refactor will touch the fewest number of files/modules possible.

### CODE REVIEW

File: doc/uml/wepClass.uml [hunk 1]
```diff
@@ -3,4 +3,7 @@ class DonutImageCheck
 class DonutDetector
 class ParamReader
 class WfEstimator
+class Image
+class Instrument
+ParamReader <|-- Instrument
```
Comment (teweitsai) [inline doc/uml/wepClass.uml]: I do not think this is correct:

```
ParamReader <|-- Instrument
```

Instead, it is:

```
Instrument *-- ParamReader
```

File: doc/uml/centroidClass.uml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+@startuml
+CentroidDefault <|-- CentroidRandomWalk
+CentroidDefault <|-- CentroidOtsu
+CentroidDefault <|-- CentroidConvolveTemplate
+CentroidFindFactory ..> CentroidRandomWalk
+CentroidFindFactory ..> CentroidOtsu
+CentroidFindFactory ..> CentroidConvolveTemplate
+CentroidConvolveTemplate *-- CentroidRandomWalk
+@enduml
```
Comment (jbkalmbach) [inline doc/uml/centroidClass.uml]: All these UML files should have an empty line at the end.

File: doc/uml/wepClass.uml (hunk 1 above)
Comment (jfcrenshaw) [reply doc/uml/wepClass.uml]: Thanks for this fix! I have pushed this change, and will rebase before the final merge 🙏

Comment (teweitsai) [reply doc/uml/wepClass.uml]: The updated commit, 8abcdc9, is:

```
ParamReader *-- Instrument
```
But it should be:

```
Instrument *-- ParamReader
```

Please help to update. Thanks!

Comment (jfcrenshaw) [reply doc/uml/wepClass.uml]: Ahh yes — sorry for that mistake 🤦‍♂️

**Review by jbkalmbach — APPROVED** [review]
Looks good to me. @teweitsai suggested this on a different pull request but maybe so we don't have to update the other packages more than once we can do all this refactor on a `develop-refactor` branch or something?

**Comment by jfcrenshaw** [comment]
Okay, I created a `develop-refactor` branch where I will now merge this PR. I will then develop PR 3 against that branch.

**Pull request merged**